src/parquet/file/reader-internal.cc

```diff
@@ -174,15 +174,13 @@ std::unique_ptr<PageReader> SerializedRowGroup::GetColumnPageReader(int i) {
   }
 
   int64_t bytes_to_read = col.meta_data.total_compressed_size;
-  std::shared_ptr<Buffer> buffer = source_->ReadAt(col_start, bytes_to_read);
+  std::unique_ptr<InputStream> stream;
 
-  if (buffer->size() < bytes_to_read) {
-    throw ParquetException("Unable to read column chunk data");
-  }
+  stream = properties_.GetInputStream();
+  stream->Initialize(source_, col_start, bytes_to_read);
```

Comment:
How about:

```c++
stream = properties_.GetStream(source_, col_start, bytes_to_read);
```

I haven't reviewed lower yet though to see if this might be a problem